backend/layers/processing/upload_failures/app.py

```diff
@@ -74,24 +121,46 @@ def get_failure_slack_notification_message(dataset_id):
                     "type": "mrkdwn",
                     "text": f"Dataset processing job failed! @sc-oncall-eng\n"
                     f"*Owner*: {collection_owner}\n"
-                    f"*Collection*: https://cellxgene.cziscience.com/collections/{collection_id}\n"
+                    f"*Collection Version URL*: {collection_version_url}\n"
+                    f"*Batch Job ID*: <{batch_url}|{job_id}>\n"
+                    f"*Step Function ARN*: <{step_function_url}|{execution_arn}>\n"
+                    f"*Error Step*: {step_name}\n"
+                    f"*Dataset ID*: {dataset_id}\n"
                     f"*Processing Status*:\n",
                 },
             },
             {
                 "type": "section",
                 "text": {
                     "type": "mrkdwn",
-                    "text": f"```{json.dumps(processing_status, cls=CustomJSONEncoder, indent=2, sort_keys=True)}```",
+                    "text": f"```{processing_status}```",
                 },
             },
         ]
     }
-    return format_failed_batch_issue_slack_alert(data)
+    return data
 
 
-def trigger_slack_notification(dataset_id):
-    data = get_failure_slack_notification_message(dataset_id)
-    logger = logging.getLogger(__name__)
+def trigger_slack_notification(
+    dataset_id: Optional[str],
+    collection_version_id: Optional[str],
+    step_name: Optional[str],
+    job_id: Optional[str],
+    aws_region: str,
+    execution_arn: str,
+) -> None:
+    data = get_failure_slack_notification_message(
+        dataset_id, collection_version_id, step_name, job_id, aws_region, execution_arn
+    )
     logger.info(data)
     notify_slack(data)
+
+
+def cleanup_artifacts(dataset_id: str, error_step_name: str) -> None:
+    """Clean up artifacts depending on error; default to full clean-up if error step is unknown"""
+    object_key = os.path.join(os.environ.get("REMOTE_DEV_PREFIX", ""), dataset_id).strip("/")
+    if not error_step_name or error_step_name == "download-validate":
+        delete_many_from_s3(os.environ["ARTIFACT_BUCKET"], object_key)
+    if not error_step_name or error_step_name == "cxg":
+        cellxgene_bucket = os.getenv("CELLXGENE_BUCKET", default=f"hosted-cellxgene-{os.environ['DEPLOYMENT_STAGE']}")
+        delete_many_from_s3(cellxgene_bucket, object_key)
```

Comment:
if we're going to clean up artifacts for instances of failed dataset processing, shouldn't we delete _all_ artifacts _every_ time?

Reply:
I would think we'd either want to delete all associated assets or none.

Reply:
note this also means cleaning up the assets from the public-access datasets bucket as well

Reply:
Ah, I'm seeing that this is appearing as added code because you wrapped it in a method; this functionality was all as-is prior to your changes. Hm. I should have added the removal of assets from the public-access datasets bucket myself. Can I get you to introduce that fix? 🙏

Reply:
in a separate PR

Reply:
https://github.com/chanzuckerberg/single-cell-data-portal/pull/4944